plugin/trino-google-sheets/src/main/java/io/trino/plugin/google/sheets/SheetsClient.java

```diff
@@ -235,16 +271,22 @@ private List<List<Object>> readAllValuesFromSheetExpression(String sheetExpressi
     {
         try {
             // by default loading up to 10k rows from the first tab of the sheet
-            String defaultRange = "$1:$10000";
-            String[] tableOptions = sheetExpression.split("#");
+            String defaultRange = DEFAULT_RANGE;
+            String[] tableOptions = sheetExpression.split(RANGE_SEPARATOR);
             String sheetId = tableOptions[0];
             if (tableOptions.length > 1) {
                 defaultRange = tableOptions[1];
             }
             log.debug("Accessing sheet id [%s] with range [%s]", sheetId, defaultRange);
-            return sheetsService.spreadsheets().values().get(sheetId, defaultRange).execute().getValues();
+            List<List<Object>> values = sheetsService.spreadsheets().values().get(sheetId, defaultRange).execute().getValues();
+            if (values == null) {
+                throw new TrinoException(SHEETS_TABLE_LOAD_ERROR, "No non-empty cells found in sheet: " + sheetExpression);
+            }
+            return values;
         }
         catch (IOException e) {
+            // TODO: improve error to a {Table|Sheet}NotFoundException
+            // after confirming the backwards incompatible error code change from 2->5 is OK
```

Comment:
what 2 and 5 are referring to?

Reply:
The error codes for: SHEETS_UNKNOWN_TABLE_ERROR and NOT_FOUND respectively. changed the comment to use the names instead of the codes to be clearer.